honeycombio/honeycomb-opentelemetry-dotnet
# Add grpc smoke tests

~**Do not merge**  Depends on #263 being merged first~ 

## Which problem is this PR solving?
Adds smoke tests for the example `aspnetcore` example app over grpc. These will eventually be used for CI smoke tests to ensure our core functionality works as expected.
- Closes #233 

## Short description of the changes
- Added `app-sdk-grpc` service to `smoke-tests/docker-compose.yml` that starts the example app with the traces endpoint configured to send traces over grpc to the collector and set the protocol to `grpc`
- Added bats tests in `smoke-tests/smoke-sdk-http.bats` that test manual instrumentation for traces

## How to test
1. Fetch this branch locally and check it out
2. `cd smoke && bats ./smoke-sdk-grpc.bats`

You should see two passing tests

## Out of scope
- Do we want to add any metrics or logs tests?
